src/router/outlet.ts

```diff
@@ -47,19 +48,25 @@ export class Outlet {
 
       if (this.cmp !== currentCmp) {
         this.vcr.clear();
-        this.vcr.createComponent(currentCmp, {
+        this.cmpRef = this.vcr.createComponent(currentCmp, {
           injector,
           environmentInjector,
         });
         this.cmp = currentCmp;
+      } else {
+        this.cmpRef?.changeDetectorRef.markForCheck();
```

Comment:
when this effect reruns for the same component for whatever reason, we'll mark this cmpRef for check just for guarantee-ness.